db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ExplainNode.java

```diff
@@ -104,6 +115,49 @@ public void optimizeStatement() throws StandardException {
             getCompilerContext().setDataSetProcessorType(DataSetProcessorType.FORCED_SPARK);
         }
         node.optimizeStatement();
+
+        // collect tables and columns that are missing statistics only for splice explain
+        // showNoStatsObjects == false for all kinds of spark explain
+        if (!showNoStatsObjects)
+            return;
```

Comment:
this "early out" somehow looks dangerous to me. someone might miss that and add something to the end of the function, not seeing most of this function is only executed if showNoStatsObjects.
i'd rather do
```
if( showNoStatsObjects )
{ 
  // code
}
```

or better - extract to a function

if( showNoStatsObjects )
  noStatsColumns = collectMissingStatisticsColumns();

Reply:
Done.